src/lib/topic-manager/code-analyzer/index.ts

```diff
@@ -0,0 +1,79 @@
+import * as core from '@actions/core';
+import { extractJson } from 'src/helpers';
+import { addQueue, QueueTaskHandler } from 'src/lib/topic-manager/code-analyzer/queue';
+import { AiResponse, TaskResult } from '../../openai/interfaces';
+import { RunnerResult, RunnerResultSuccess } from '../../openai/openai.service';
+import { createPrompt, getAdditionalInstructions } from '../../openai/prompt';
+import { TopicManager } from '../topic-manager';
+import { PullRequestDetails, ValidTopic } from '../topic-manager.interface';
+
+export class CodeAnalyzer {
+  constructor(
+    private readonly prDetails: PullRequestDetails,
+    private readonly manager: TopicManager
+  ) {}
+
+  private safeReturnDto(d: RunnerResult | null): AiResponse {
+    if (!d?.success) return { success: false, reviews: [], topicId: null };
+    const { content } = d as RunnerResultSuccess;
+    const reviews = extractJson<AiResponse>(content);
+    return { success: !!reviews, ...((reviews || {}) as AiResponse) };
+  }
+
+  private preparePrompts(contentList: ValidTopic[]) {
+    return contentList.map(item => ({
+      ...item,
+      prompt: createPrompt(item)
+    }));
+  }
+
+  private getLanguageInstructions(language = 'pt-br') {
+    return getAdditionalInstructions(language);
+  }
+
+  private createAnalysisTask(prompt: ValidTopic, additionalInstructions: string): QueueTaskHandler<TaskResult> {
+    return async ({ jobId = '--' }) => {
+      const threadId = prompt.topicId;
+      return this.processThread(jobId, prompt, threadId, additionalInstructions);
+    };
+  }
+
+  private async processThread(jobId: string, prompt: ValidTopic, threadId: string, additionalInstructions: string): Promise<TaskResult> {
+    core.info(`Processing job ${jobId} ${prompt.filename} threadId: ${threadId}`);
+
+    const messageCreated = await this.manager.createTopicMessage(threadId, {
+      role: 'user',
+      content: prompt?.prompt || '--',
```

Comment:
⚠️
O código `prompt?.prompt || '--'` assume que `prompt.prompt` pode ser nulo ou indefinido. Certifique-se de que essa situação é possível e, se não for desejável, implemente uma validação mais estrita.
Manipular valores potencialmente nulos sem uma verificação explícita pode levar a comportamentos inesperados ou erros em tempo de execução.